test/test_linear_space_lcs.py

```diff
@@ -0,0 +1,60 @@
+import os
+import unittest
+
+from approximate_string_matching import linear_space_lcs, lcs
+from generator import rand
+
+LCS_ALGORITHMS = [
+    linear_space_lcs.linear_space_lcs,
+]
+
+class TestLinearSpaceLongestCommonSubsequence(unittest.TestCase):
+  run_large = unittest.skipUnless(
+      os.environ.get('LARGE', False), 'Skip test in small runs')
+
+  """Test if substr is subsequence of the word"""
+  def is_subsequence(self, word, substr):
+    j = 0
+    for i in range(0, len(word)):
+      if word[i] == substr[j]:
+        j = j + 1
+      if j == len(substr):
+        return True
```

Comment:
Alternatively you may use:
def is_subsequence(x, y):
it = iter(y)
return all(any(c == ch for c in it) for ch in x)
Since we use the same "it" over the whole second line, it should use all letters of y only once and in proper order.

Reply:
I decided to stick with my version, I am not sure how that one works